packages/kit/src/build.ts

```diff
@@ -136,3 +136,18 @@ export function addVitePlugin (pluginOrGetter: VitePlugin | VitePlugin[] | (() =
     }
   }, options)
 }
+
+interface AddBuildPluginFactory {
+  vite?: () => VitePlugin | VitePlugin[]
+  webpack?: () => WebpackPluginInstance | WebpackPluginInstance[]
+}
+
+export function addBuildPlugin (pluginFactory: AddBuildPluginFactory, options?: ExtendConfigOptions) {
+  if (pluginFactory.vite) {
+    addVitePlugin(pluginFactory.vite, options)
+  }
+
+  if (pluginFactory.webpack) {
+    addWebpackPlugin(pluginFactory.webpack, options)
+  }
+}
```

Comment:
Just an idea: This function would also check if _current builder_ is supported or not and make a warning a build plugin is added without vite/webpack support

Reply:
I love the idea. Also maybe provide verbose warning and suggestion in module building mode?

Reply:
For module building (i guess testing is only possible) we could do if partial in either case too :)